src/colvar_gpu_support.cpp

```diff
@@ -0,0 +1,121 @@
+#include "colvar_gpu_support.h"
+#include "colvarmodule.h"
+
+#include <cstring>
+
+#if defined(__has_include)
+# if __has_include(<stacktrace>)
+#  define use_cpp_stacktrace
+#  include <stacktrace>
+# endif
+
+#endif
+
+namespace colvars_gpu {
+#if defined (COLVARS_CUDA) || defined (COLVARS_HIP)
+int gpuAssert(cudaError_t code, const char *file, int line)
+{
+  if (code != cudaSuccess) {
+    std::string error =
+      std::string("GPUassert: ") +
+      cudaGetErrorString(code) + " " + file + ", line " + cvm::to_str(line);
+#ifdef use_cpp_stacktrace
+#ifdef __cpp_lib_stacktrace
+      error += "\nBacktrace:\n" + std::to_string(std::stacktrace::current()) + "\n";
+#endif
+#endif
+    return cvm::error(error, COLVARS_ERROR);
```

Comment:
When this detects an error at the time of `colvarproxy` construction, before `colvarmodule` is allocated, `cvm::error()` triggers a segfault, and no error message is printed. Possible solutions:

- make this function avoid using cvm methods
- make the proxy constructor use a safe alternative